Take care of file rename operations in `rename` refactorings

mlangc opened the pull request:
Note that this pull request depends on https://github.com/scala-ide/scala-refactoring/pull/80 and will therefore **not compile** unless an updated version of scala-refactoring becomes available.

Fixes [#1001928](https://www.assembla.com/spaces/scala-ide/tickets/1001928#/activity/ticket:)

sschaef commented:
add to whitelist

mlangc commented:
@sschaef I can update the commit message to reference the associated refactoring commit for further information as soon as we agreed on the latter.

sschaef commented on org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/ScalaIdeRefactoring.scala:
```diff
@@ -137,9 +145,17 @@ abstract class ScalaIdeRefactoring(val getName: String, val file: ScalaSourceFil
     }
   }
 
+  private def toIPath(path: AbstractFile) = {
```
this functionality already exists in `FileUtils`

mlangc commented:
@sschaef I've moved `toIPath` to `FileUtils` and also updated the commit message to reference scala-ide/scala-refactoring@21e3ccf59123824a22b6d60e4bb991ccc58d2de9.

sschaef commented:
retest this please

sschaef commented:
LGTM

sschaef commented:
Actually, the refactoring on the type is only successful when all the relevant files are opened. Closed files are not changed. But this is a problem unrelated to this PR and just another bug in the refactoring logic.
